burn-autodiff/src/ops/tensor.rs

```diff
@@ -1372,6 +1372,28 @@ impl<B: Backend> TensorOps<ADBackendDecorator<B>> for ADBackendDecorator<B> {
             }
         }
     }
+
+    fn clamp_min<const D: usize>(
+        _tensor: <ADBackendDecorator<B> as Backend>::TensorPrimitive<D>,
+        _min: <ADBackendDecorator<B> as Backend>::FloatElem,
+    ) -> <ADBackendDecorator<B> as Backend>::TensorPrimitive<D> {
+        todo!()
+    }
+
+    fn clamp_max<const D: usize>(
+        _tensor: <ADBackendDecorator<B> as Backend>::TensorPrimitive<D>,
+        _max: <ADBackendDecorator<B> as Backend>::FloatElem,
+    ) -> <ADBackendDecorator<B> as Backend>::TensorPrimitive<D> {
+        todo!()
+    }
+
+    fn clamp<const D: usize>(
+        _tensor: <ADBackendDecorator<B> as Backend>::TensorPrimitive<D>,
+        _min: <ADBackendDecorator<B> as Backend>::FloatElem,
+        _max: <ADBackendDecorator<B> as Backend>::FloatElem,
+    ) -> <ADBackendDecorator<B> as Backend>::TensorPrimitive<D> {
+        todo!()
+    }
```

Comment:
If you look at the default implementation, the gradient should be zero when its value was clamped during the forward pass. So there are two possible implementations:

1. Don't override the operation in the burn-autodiff backend, so the backward pass will be calculated with the `mask_fill` operation.
2. Override it, and keep the input tensor as state to compute the mask during the backward pass and then use the mask_fill.

I prefer the first method, since in both cases the memory requirement is the same, and even protetially lower (bool tensor < float tensor) and the implementation is way simpler (nothing to write). In terme of computation, there are 2 (potentially inplace) operations during the forward pass, and 1 (potentially inplace) operation  during the backward pass for the first solution. The second solution has 1 operation (never inplace) during the forward pass, and two (potentially inplace) operations during the backward pass.